HealthGPT/HealthGPT/HealthGPTView.swift

```diff
@@ -6,60 +6,78 @@
                     .gesture(
                         TapGesture().onEnded {
                             UIApplication.shared.hideKeyboard()
                         }
                     )
-                MessageInputView()
-                    .environmentObject(messageManager)
-            }
-            .navigationBarTitle("HealthGPT")
-            .alert(isPresented: $showAlert) {
-                Alert(
-                    title: Text("Alert"),
-                    message: Text(alertText),
-                    dismissButton: .default(Text("OK"))
-                )
-            }
-            .onAppear {
-                updateApiKeyAndModel()
+                    .onChange(of: messages) { _ in
+                        if !gettingAnswer {
+                            getAnswer()
+                        }
+                    }
             }
+            //  create a sheet with settingsview, add a button
+            //  SettingsView(chat: $messages)
         }
     }
+    
+    private func getAnswer() {
+        _Concurrency.Task {
+            do {
+                gettingAnswer = true
+                
+                // add chat
+                let healthData = try await healthDataFetcher.fetchAndProcessHealthData()
 
-    private func updateApiKeyAndModel() {
-        guard let apiKey = try? secureStorage.retrieveCredentials(
-            "openai-api-key",
-            server: "openai.com"
-        )?.password else {
-            alertText = "Could not find a valid API key."
-            self.showAlert.toggle()
-            return
+                let generator = PromptGenerator(with: healthData)
+                let mainPrompt = generator.buildMainPrompt()
+                
+                // create full prompt
+                var fullPrompt = [Chat(role: .system, content: mainPrompt)]
+                for message in messages {
+                    fullPrompt.append(Chat(role: message.role, content: message.content))
+                }
+                
+                // add mainprompt to front of self.messages
+                let chatStreamResults = try await openAPIComponent.queryAPI(withChat: fullPrompt)
+                
+                for try await chatStreamResult in chatStreamResults {
+                    for choice in chatStreamResult.choices {
+                        if messages.last?.role == .assistant {
+                            let previousChatMessage = messages.last ?? Chat(role: .assistant, content: "")
+                            messages[messages.count - 1] = Chat(
+                                role: .assistant,
+                                content: (previousChatMessage.content ?? "") + (choice.delta.content ?? "")
+                            )
+                        } else {
+                            messages.append(Chat(role: .assistant, content: choice.delta.content ?? ""))
+                        }
+                    }
+                }
```

Comment:
It would be great if we can abstract this functionality in a dedicated Spezi Component to ensure that we can encapsulate that functionality and make sure that we separate view logic from the computations happening in HealthGPT.
In the process addresses two issues that we currently encounter in the code above:
1. Every time the user types a message, we query all the health data and generate a completely new chat based on all that health data. In an ideal situation we only do this once, e.g. when the component is generated (we can use the configure method for that) and then rely on that chat going forward and only append newer messages as we get them from the mode.
2. Ensure that we can eventually use Spezi to keep track of that component and maybe even move that functionality out in a new module or some reusable code that we could share across LLM on FHIR and this app (we have mostly copied code so far).

LLM on FHIR provides you a good starting point to see how you can do that, e.g. the [`FHIRResourceSummary`](https://github.com/StanfordBDHG/LLMonFHIR/blob/main/LLMonFHIR/FHIR%20Display/FHIRResourceSummary.swift#L21) type.

The steps would be the following:
1. Create a new component and name it, e.g., `HealthDataInterpreter` by following the [Spezi documentation](https://swiftpackageindex.com/stanfordspezi/spezi/0.5.3/documentation/spezi/modulecapabilities) and taking a look at the the [`FHIRResourceSummary`](https://github.com/StanfordBDHG/LLMonFHIR/blob/main/LLMonFHIR/FHIR%20Display/FHIRResourceSummary.swift#L21) type.
2. The component can expose a `messages: [Chat]` property that can then be used to feed it in the ChatView like we do here and then uses the HealthDataFetcher and Prompt Generator internally.
3. We add this component in the App Delegate and therefore make it available in the SwiftUI environment to use it there in this view but also in the settings view (so we don't need to explicitly need to pass in the chat and we could e.g. also remove any health data that we retrieved from HealthKit in the same process).